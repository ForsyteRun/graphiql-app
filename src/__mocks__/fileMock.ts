import { DevelopersData, Role } from '../types';
import photo from './../assets/img/forsyte.jpg';

export const forsyte: DevelopersData = {
  name: 'Forsyte Alex',
  image: photo,
  role: Role.lead,
  github: 'forsyterun',
  about:
    'I`m is a Ukrainian singer, producer, and DJ known for his contributions to the pop and electronic music scenes. Born on October 17, 1988, in Chelyabinsk, Russia,',
  contribution: [
    'JavaScript',
    'Python',
    'Java',
    'C++',
    'C#',
    'TypeScript',
    'Ruby',
    'Swift',
    'Go',
    'Kotlin',
    'Rust',
    'PHP',
    'HTML',
    'CSS',
  ],
};

export const Katsiaryna: DevelopersData = {
  name: 'Katsiaryna',
  image: photo,
  role: Role.developer,
  github: 'Katsiaryna',
  about:
    'I`m is a Ukrainian singer, producer, and DJ known for his contributions to the pop and electronic music scenes. Born on October 17, 1988, in Chelyabinsk, Russia,',
  contribution: [
    'JavaScript',
    'Python',
    'Java',
    'C++',
    'C#',
    'TypeScript',
    'Ruby',
    'Swift',
    'Go',
    'Kotlin',
    'Rust',
    'PHP',
    'HTML',
    'CSS',
  ],
};

export const nastia: DevelopersData = {
  name: 'nyurasheva',
  image: photo,
  role: Role.developer,
  github: 'nyurasheva',
  about:
    'I`m is a Ukrainian singer, producer, and DJ known for his contributions to the pop and electronic music scenes. Born on October 17, 1988, in Chelyabinsk, Russia,',
  contribution: [
    'JavaScript',
    'Python',
    'Java',
    'C++',
    'C#',
    'TypeScript',
    'Ruby',
    'Swift',
    'Go',
    'Kotlin',
    'Rust',
    'PHP',
    'HTML',
    'CSS',
  ],
};
