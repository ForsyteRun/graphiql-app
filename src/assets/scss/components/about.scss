/* about */

.about {
  width: 100%;
  background: $background;
  z-index: 100;

  &__title {
    @include font(
      $family: $font-family-title,
      $size: $font-size-32,
      $weight: bold
    );
    color: $button;
    padding: 70px 0 100px;
  }

  &__content {
    background: $text-color;
    width: 100%;
    padding: 0 0 5px;

    p {
      color: $background;
      text-align: left;
    }
  }

  &__item {
    @include flex($justify: stretch);
    width: 1030px;
    height: auto;
    min-height: 280px;
    margin: 0 auto 90px;
    position: relative;
    border: 1px solid #ffffff40;
    background: #1e1e1e;
    transition: $transition;

    &:hover {
      background: #454545;
      cursor: pointer;

      .about__number {
        top: -40px;
      }
    }
  }

  &__images,
  &__descriptions {
    @include flexColumn($align: stretch);
    text-align: left;
    opacity: 0;
    transition:
      opacity 1s ease,
      transform 1s ease;

    &.visible {
      opacity: 1;
      transform: none;
    }
  }

  &__descriptions {
    width: 60%;
    padding: 0 40px 20px 20px;
    gap: 25px;
    text-shadow: 2px 2px 2px rgba(0, 0, 0, 0.5);
    transform: translateY(50%) translateX(50%);

    h2 {
      @include font(
        $family: $font-family-title,
        $size: $font-size-28,
        $weight: bold
      );
      @include line();
      color: $button-second;
    }
  }

  &__images {
    width: 40%;
    transform: translateY(-50%) translateX(-50%);
  }

  &__img {
    height: 280px;
    width: 100%;
    background-size: cover;
    background-position: top center;
    box-shadow: 0px 3px 4px 0px #00000033;
    position: relative;
    top: -50px;
    left: -50px;
  }

  &__number {
    @include font(
      $family: $font-family-title,
      $size: $font-size-19,
      $weight: bold
    );
    position: absolute;
    width: 2em;
    height: 2em;
    padding: 10px;
    top: -20px;
    right: 22px;
    color: $links;
    border: 2px solid $links;
    border-radius: 50%;
    text-align: center;
    transition: $transition;
  }

  &__svg {
    width: 100%;
    margin-top: -1px;
  }
}
