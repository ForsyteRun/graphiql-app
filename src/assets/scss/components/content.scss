.welcome-list {
  align-self: flex-end;

  & li {
    align-self: flex-end;
    font-size: 1.5rem;
    font-weight: 700;

    &::before {
      display: none;

      @include media('xs') {
        content: '\2022';
        color: $blue-color;
        font-weight: bold;
        display: inline-block;
        width: 1em;
        margin-left: -1em;
      }
    }

    @include media('xl') {
      font-size: clamp(14px, 2vw, 1.5rem);
    }

    @include media('xs') {
      font-size: 1.5rem;
      margin-bottom: 0.5rem;
    }

    span {
      display: inline-block;
      font-weight: 400;
      margin: 0.7rem 2px 0.7rem;

      @include media('xl') {
        margin: 0.5rem 2px 0.5rem;
      }

      @include media('s') {
        margin: 0.2rem 2px 0.2rem;
      }

      @include media('xs') {
        display: none;
      }
    }
  }
}
