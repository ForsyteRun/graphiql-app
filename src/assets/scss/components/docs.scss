aside {
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 100%;
  background: #999999;
  transition: all 1s ease;
  & .docs-container {
    height: 100%;
    width: 100%;
    overflow-y: auto;
  }
  & .docs-button {
    position: absolute;
    left: 0;
    bottom: 20%;
    background: $links;
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    transform: rotate(270deg) translateY(-178%);
    padding: 1rem 3rem;
  }
  & .docs-details {
    @include flexColumn(flex-start, flex-start);
    row-gap: 1rem;
    padding: 1rem;
    &__type {
      & span {
        color: rgb(0, 255, 55);
      }
    }
    &__description {
      & span {
        color: rgb(214, 209, 232);
      }
    }
  }
}

.openModal {
  left: 60%;
  transition: all 1s ease;
}
