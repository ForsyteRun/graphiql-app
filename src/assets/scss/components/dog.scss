/* dog */

.dog {
  background: $background;
  width: 100%;
  margin-top: -1px;

  &__content {
    @include flex();
    padding-bottom: 40px;
  }

  &__img {
    flex: 1;
    transform: translateY(50px);
    transition:
      opacity 1s ease,
      transform 1s ease;

    &.visible {
      transform: translateY(0);
    }
  }

  &__description {
    @include flexColumn($align: flex-start);
    flex: 1;
    gap: 20px;
    padding: 0 2rem;
  }

  &__title {
    @include font(
      $family: $font-family-title,
      $size: $font-size-32,
      $weight: bold
    );
  }

  &__buttons {
    @include flex();
    gap: 30px;
  }
}
