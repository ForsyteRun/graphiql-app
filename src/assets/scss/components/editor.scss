/* editor */

.main__title {
  color: #9ffffb;
}

.api,
.headers,
.variables {
  @include flex($justify: space-between, $align: end);
  gap: 2rem;
  margin: 0 0 20px;
  color: $background;

  &__content {
    @include flexColumn($align: stretch);
    flex: 1 1 auto;
    gap: 0.5rem;
  }

  &__endpoint-text {
    white-space: nowrap;
  }

  &__input {
    width: 100%;
    height: 36px;
    border: $border;
    border-radius: 8px;
    padding: 0 8px;
  }

  &__label {
    min-width: 130px;
    text-align: left;
    color: #9ffffb;
  }
}

.editor {
  padding-bottom: 30px;

  &__wrapper {
    @include flex($justify: space-between, $align: baseline, $wrap: wrap);
    border: solid 10px #0d0c18;
    border-radius: 4px;
    overflow: hidden;
    margin: 0;

    * {
      font-family: menlo, monospace;
      font-size: 1rem;
      line-height: 150%;
      padding: 0;
      margin: 0;
    }
  }

  &__left,
  &__right {
    display: flex;
    flex-direction: column;
    flex: 1;
    border-left: solid 1px #2b2b2b;
    border-bottom: solid 1px #2b2b2b;
  }

  &__left {
    .editor__content {
      background: #1f1f1f47;
    }
  }

  &__right {
    border-right: solid 1px #2b2b2b;
  }

  &__header {
    @include flex($justify: space-between);
    min-height: 52px;
    text-align: left;
    padding: 10px 15px;
    background: #181818;
    border-top: solid 1px #2b2b2b;
    border-bottom: solid 1px #2b2b2b;
    border-radius: 2px 2px 0 0;
    color: #ffffff99;
  }

  &__content {
    @include flex($align: stretch);
    background: #1f1f1f;
    color: #739db3;
    flex: 1 0 auto;
    position: relative;
  }

  &__numbers {
    @include flexColumn();
    color: #999;
    background: #181818;
    font-family: menlo, monospace;
    padding: 10px 10px;
    height: 100%;
    line-height: 150%;
    text-align: right;
  }

  &__text {
    @include flexColumn($align: stretch);
    width: 100%;
  }

  &__query {
    height: 100%;
    border: none;
    background: #2c2e2f;
    text-align: left;
    color: $background;

    &:hover {
      border: none;
      box-shadow: none;
    }
  }

  &__response {
    text-align: left;
  }

  &__buttons {
    @include flex();
    gap: 10px;
  }

  &__play,
  &__play:hover {
    min-width: 30px;
    height: 30px;
    background: url(/src/assets/svg/play.svg) no-repeat center center/cover;
  }

  &__color,
  &__color:hover {
    min-width: 30px;
    height: 30px;
    background: url(/src/assets/svg/colors.svg) no-repeat center center/cover;
  }
}
