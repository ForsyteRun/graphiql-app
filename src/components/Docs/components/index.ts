export { default as FieldSchema } from './FieldSchema';
export { default as DetailFieldSchema } from './DetailFieldSchema';
