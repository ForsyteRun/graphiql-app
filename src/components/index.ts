export { default as AppRouter } from './AppRouter';
export { default as GraphiQL } from './GraphiQL';
export { default as LoginForm } from './LoginForm';
export { default as Logo } from './Logo';
export { default as RegistrationForm } from './RegistrationForm';
