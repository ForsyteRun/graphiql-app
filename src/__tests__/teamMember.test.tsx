import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Katsiaryna, forsyte, nastia } from '../__mocks__/fileMock';
import { TeamMember } from '../components';

describe('TeamMember component', () => {
  it('should be equal forsyte snapshot', () => {
    const { asFragment } = render(<TeamMember developer={forsyte} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be equal forsyte snapshot', () => {
    const { asFragment } = render(<TeamMember developer={Katsiaryna} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should be equal forsyte snapshot', () => {
    const { asFragment } = render(<TeamMember developer={nastia} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
