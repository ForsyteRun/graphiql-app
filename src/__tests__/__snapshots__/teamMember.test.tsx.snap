// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMember component should be equal forsyte snapshot 1`] = `
<DocumentFragment>
  <div
    class="team"
  >
    <div
      class="team__info"
    >
      <img
        alt="Forsyte Alex"
        class="team__photo"
      />
      <div
        class="team__details"
      >
        <h3
          class="team__name"
        >
          Forsyte Alex
        </h3>
        <h4
          class="team__role"
        >
          Team Lead
        </h4>
        <a
          class="team__github"
          href="https://github.com/forsyterun"
          rel="noreferrer"
          target="_blank"
          title="Forsyte Alex GitHub"
        >
          forsyterun
        </a>
        <div>
          <h4
            class="team__contribution_title"
          >
            Contribution
          </h4>
          <ul
            class="team__contribution_list"
          >
            <li>
              JavaScript
            </li>
            <li>
              Python
            </li>
            <li>
              Java
            </li>
            <li>
              C++
            </li>
            <li>
              C#
            </li>
            <li>
              TypeScript
            </li>
            <li>
              Ruby
            </li>
            <li>
              Swift
            </li>
            <li>
              Go
            </li>
            <li>
              Kotlin
            </li>
            <li>
              Rust
            </li>
            <li>
              PHP
            </li>
            <li>
              HTML
            </li>
            <li>
              CSS
            </li>
          </ul>
        </div>
      </div>
    </div>
    <h4
      class="team__about_title"
    >
      About
    </h4>
    <p
      class="team__about"
    >
      I\`m is a Ukrainian singer, producer, and DJ known for his contributions to the pop and electronic music scenes. Born on October 17, 1988, in Chelyabinsk, Russia,
    </p>
  </div>
</DocumentFragment>
`;

exports[`TeamMember component should be equal forsyte snapshot 2`] = `
<DocumentFragment>
  <div
    class="team"
  >
    <div
      class="team__info"
    >
      <img
        alt="Katsiaryna"
        class="team__photo"
      />
      <div
        class="team__details"
      >
        <h3
          class="team__name"
        >
          Katsiaryna
        </h3>
        <h4
          class="team__role"
        >
          Developer
        </h4>
        <a
          class="team__github"
          href="https://github.com/Katsiaryna"
          rel="noreferrer"
          target="_blank"
          title="Katsiaryna GitHub"
        >
          Katsiaryna
        </a>
        <div>
          <h4
            class="team__contribution_title"
          >
            Contribution
          </h4>
          <ul
            class="team__contribution_list"
          >
            <li>
              JavaScript
            </li>
            <li>
              Python
            </li>
            <li>
              Java
            </li>
            <li>
              C++
            </li>
            <li>
              C#
            </li>
            <li>
              TypeScript
            </li>
            <li>
              Ruby
            </li>
            <li>
              Swift
            </li>
            <li>
              Go
            </li>
            <li>
              Kotlin
            </li>
            <li>
              Rust
            </li>
            <li>
              PHP
            </li>
            <li>
              HTML
            </li>
            <li>
              CSS
            </li>
          </ul>
        </div>
      </div>
    </div>
    <h4
      class="team__about_title"
    >
      About
    </h4>
    <p
      class="team__about"
    >
      I\`m is a Ukrainian singer, producer, and DJ known for his contributions to the pop and electronic music scenes. Born on October 17, 1988, in Chelyabinsk, Russia,
    </p>
  </div>
</DocumentFragment>
`;

exports[`TeamMember component should be equal forsyte snapshot 3`] = `
<DocumentFragment>
  <div
    class="team"
  >
    <div
      class="team__info"
    >
      <img
        alt="nyurasheva"
        class="team__photo"
      />
      <div
        class="team__details"
      >
        <h3
          class="team__name"
        >
          nyurasheva
        </h3>
        <h4
          class="team__role"
        >
          Developer
        </h4>
        <a
          class="team__github"
          href="https://github.com/nyurasheva"
          rel="noreferrer"
          target="_blank"
          title="nyurasheva GitHub"
        >
          nyurasheva
        </a>
        <div>
          <h4
            class="team__contribution_title"
          >
            Contribution
          </h4>
          <ul
            class="team__contribution_list"
          >
            <li>
              JavaScript
            </li>
            <li>
              Python
            </li>
            <li>
              Java
            </li>
            <li>
              C++
            </li>
            <li>
              C#
            </li>
            <li>
              TypeScript
            </li>
            <li>
              Ruby
            </li>
            <li>
              Swift
            </li>
            <li>
              Go
            </li>
            <li>
              Kotlin
            </li>
            <li>
              Rust
            </li>
            <li>
              PHP
            </li>
            <li>
              HTML
            </li>
            <li>
              CSS
            </li>
          </ul>
        </div>
      </div>
    </div>
    <h4
      class="team__about_title"
    >
      About
    </h4>
    <p
      class="team__about"
    >
      I\`m is a Ukrainian singer, producer, and DJ known for his contributions to the pop and electronic music scenes. Born on October 17, 1988, in Chelyabinsk, Russia,
    </p>
  </div>
</DocumentFragment>
`;
